cpp/src/arrow/compute/kernels/temporal_internal.h

```diff
@@ -269,6 +283,18 @@ struct TemporalComponentExtractBase<Op, std::chrono::seconds, Time32Type, OutTyp
   }
 };
 
+template <template <typename...> class Op, typename OutType>
+struct TemporalComponentExtractBase<Op, std::chrono::seconds, DurationType, OutType> {
+  template <typename OptionsType>
+  static Status ExecWithOptions(KernelContext* ctx, const OptionsType* options,
+                                const ExecSpan& batch, ExecResult* out) {
+    using ExecTemplate = Op<std::chrono::seconds, NonZonedLocalizer, DurationType>;
```

Comment:
Here the `DurationType` is explicitly specified as the third type parameter so it goes to the corresponding specification.